account/src/main/java/com/user/account/controller/AllControlers.java

```diff
@@ -50,55 +39,55 @@ public String home() {
 		
 		
 		@PostMapping("/debit/{accountNumber}")
-		public void debit(@PathVariable String accountNumber, @RequestBody String amount) {
-
-			long userAccountNumber = Long.parseLong(accountNumber);
-
-			long debitAmount = Long.parseLong(amount);
+		public ResponseEntity<String> debit(@PathVariable Long accountNumber, @RequestBody Long amount) {
 			
-			User user = accountService.getUser(userAccountNumber);
+			User user = accountService.getUser(accountNumber);
 			
-			if ( debitAmount <= user.getAccountBalance() ) {
+			if ( amount.compareTo(user.getAccountBalance()) <= 0 ) {
 				
-				user.incrementAccountBalance(-debitAmount) ;
+				user.incrementAccountBalance(-amount) ;
```

Comment:
Have a `decrementAccountBalance` function to debit amount instead of using the same `incrementAccountBalance` function for debit and credit

Reply:
To avoid code redundancy decrementAccountBalance is not created.

Reply:
decrementAccountBalance is more readable and hence should be preferred.

You can save code redundancy by call `incrementAccountBalance (-amount)` inside `decrementAccountBalance` method

Reply:
Written separate incementAccountBalance and decrementAccountBalance method